tests/ignite/distributed/test_auto.py

```diff
@@ -57,6 +79,8 @@ def _test(data):
             sampler = None
         elif sampler_name == "WeightedRandomSampler":
             sampler = WeightedRandomSampler(weights=torch.ones(100), num_samples=100)
+        elif sampler_name == "DistributedSampler":
+            sampler = DistributedSampler(data, num_replicas=ws, rank=idist.get_rank())
```

Comment:
@schuhschuh I think we have to update also the check below:
```python
        if isinstance(data, IterableDataset):
            sampler_type = _InfiniteConstantSampler
        elif ws > 1:
            sampler_type = DistributedSampler if sampler is None else DistributedProxySampler
        else:
            sampler_type = RandomSampler if sampler is None else type(sampler)
```
?

Reply:
I thought so at first, too. But looking more closely, the `else` branch uses `type(sampler)` as the expected type, which in case of a user provided distributed sampler should be the type of the user provided sampler.

I also considered to disallow the use of a distributed sampler in `auto_dataloader()` when world size is 1 as in the `no_dist` test case. But given that a distributed sampler with world size 1 and rank 0 should be fine, I think it's best to allow this use case even if the data loading is not actually distributed?

Reply:
Ah, I see what you mean. The `ws > 1` case rather, where type no should be `DistributedSampler` instead of `DistributedProxySampler`. Will fix the test.

Reply:
> I also considered to disallow the use of a distributed sampler in auto_dataloader() when world size is 1 as in the no_dist test case. But given that a distributed sampler with world size 1 and rank 0 should be fine, I think it's best to allow this use case even if the data loading is not actually distributed?

Yes, I agree, let's allow this use case.